src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryInstrumenter.cs

```diff
@@ -135,55 +145,93 @@ internal static class RedisProfilerEntryToActivityConverter
-                    activity.SetTag(SemanticConventions.AttributePeerService, command.EndPoint.ToString());
-                }
+                var ip = ipEndPoint.Address.ToString();
+                var port = ipEndPoint.Port;
+
+                activity?.SetTag(SemanticConventions.AttributeNetPeerIp, ip);
+                activity?.SetTag(SemanticConventions.AttributeNetPeerPort, port);
+
+                meterTags?.Add(SemanticConventions.AttributeNetPeerIp, ip);
+                meterTags?.Add(SemanticConventions.AttributeNetPeerPort, port);
+            }
+            else if (command.EndPoint is DnsEndPoint dnsEndPoint)
+            {
+                var host = dnsEndPoint.Host;
+                var port = dnsEndPoint.Port;
+
+                activity?.SetTag(SemanticConventions.AttributeNetPeerName, host);
+                activity?.SetTag(SemanticConventions.AttributeNetPeerPort, port);
+
+                meterTags?.Add(SemanticConventions.AttributeNetPeerName, host);
+                meterTags?.Add(SemanticConventions.AttributeNetPeerPort, port);
             }
+            else
+            {
+                var service = command.EndPoint.ToString();
+
+                activity?.SetTag(SemanticConventions.AttributePeerService, service);
+                meterTags?.Add(SemanticConventions.AttributePeerService, service);
+            }
+        }
 
-            activity.SetTag(StackExchangeRedisConnectionInstrumentation.RedisDatabaseIndexKeyName, command.Db);
+        var db = command.Db;
+        activity?.SetTag(SemanticConventions.AttributeDbRedisDatabaseIndex, db);
+        meterTags?.Add(SemanticConventions.AttributeDbRedisDatabaseIndex, db);
 
-            // TODO: deal with the re-transmission
-            // command.RetransmissionOf;
-            // command.RetransmissionReason;
+        // TODO: deal with the re-transmission
+        // command.RetransmissionOf;
+        // command.RetransmissionReason;
 
+        if (activity?.IsAllDataRequested ?? false)
+        {
             var enqueued = command.CommandCreated.Add(command.CreationToEnqueued);
             var send = enqueued.Add(command.EnqueuedToSending);
             var response = send.Add(command.SentToResponse);
+            var completion = send.Add(command.ResponseToCompletion);
 
             if (options.EnrichActivityWithTimingEvents)
             {
                 activity.AddEvent(new ActivityEvent("Enqueued", enqueued));
                 activity.AddEvent(new ActivityEvent("Sent", send));
                 activity.AddEvent(new ActivityEvent("ResponseReceived", response));
+                activity.AddEvent(new ActivityEvent("Completion", completion));
             }
 
             options.Enrich?.Invoke(activity, command);
         }
 
-        activity.Stop();
+        if (metrics.Enabled && meterTags is TagList meterTagList)
+        {
+            metrics.QueueHistogram.Record(command.EnqueuedToSending.TotalSeconds, meterTagList.ToArray());
```

Comment:
why `ToArray()` is needed?

Reply:
Good catch, I added it to a unit test validation and forgot this ToArray(). Removed.